external/src/main/java/com/redhat/cloud/policies/engine/lightweight/LightweightEngineImpl.java

```diff
@@ -229,61 +297,73 @@ private Set<Alert> fireTriggers(Event event) {
+                    LOGGER.debugf("Starting the trigger conditions evaluation");
+                    Set<ConditionEval> conditionEvals = new HashSet<>();
+                    for (Condition condition : fullTrigger.getConditions()) {
+                        switch (condition.getType()) {
+                            case EVENT:
+                                evaluateCondition(fullTrigger.getTrigger(), (EventCondition) condition, event).ifPresent(conditionEvals::add);
+                                break;
+                            default:
+                                LOGGER.warnf("Unsupported condition type %s", condition);
+                                break;
+                        }
                     }
-                }
-                // The trigger should always have exactly one dampening.
-                Dampening dampening = fullTrigger.getDampenings().get(0);
-                performDampening(dampening, fullTrigger.getTrigger(), conditionEvals);
-                if (dampening.isSatisfied()) {
-                    LOGGER.debugf("Dampening satisfied");
-                    Alert alert = new Alert(event.getTenantId(), fullTrigger.getTrigger(), dampening, dampening.getSatisfyingEvals());
-                    policiesHistoryService.put(alert);
-                    if (shouldSendNotification(fullTrigger)) {
-                        alerts.add(alert);
+                    // The trigger should always have exactly one dampening.
+                    Dampening dampening = fullTrigger.getDampenings().get(0);
+                    performDampening(dampening, fullTrigger.getTrigger(), conditionEvals);
+                    if (dampening.isSatisfied()) {
+                        LOGGER.debugf("Dampening satisfied");
+                        Alert alert = new Alert(event.getTenantId(), fullTrigger.getTrigger(), dampening, dampening.getSatisfyingEvals());
+                        policiesHistoryService.put(alert);
+                        if (shouldSendNotification(fullTrigger)) {
+                            alerts.add(alert);
+                        }
+                    } else {
+                        LOGGER.debugf("Dampening not satisfied");
                     }
-                } else {
-                    LOGGER.debugf("Dampening not satisfied");
+                    dampening.reset();
                 }
-                dampening.reset();
             }
             return alerts;
         }
     }
 
     private boolean shouldSendNotification(FullTrigger fullTrigger) {
-        if (fullTrigger.getTrigger().getActions() != null) {
-            for (TriggerAction triggerAction : fullTrigger.getTrigger().getActions()) {
-                if (triggerAction.getActionPlugin().equals("email") || triggerAction.getActionPlugin().equals("notification")) {
-                    return true;
+        if (lightweightEngineConfig.isRestApiEnabled()) {
+            return true;
```

Comment:
Why it returns true without checking the action plugins of the trigger? Is this done elsewhere?